R/mod-panel-section.R

```diff
@@ -38,24 +47,58 @@ mod_panel_section_ui <- function(id){
 #' @keywords internal
 mod_panel_section_server <- function(input, output, session, synapse, syn,
                                      reviews_table) {
+  submission <- reactive({ input$submission })
+
   ## Load reviews
-  reviews <- syn$tableQuery(glue::glue("SELECT * FROM {reviews_table}"))
-  reviews <- readr::read_csv(reviews$filepath) %>%
-    dplyr::mutate(scorer = get_display_name(syn, .data$scorer))
-  
+  reviews <-reactive(pull_reviews_table(syn, reviews_table))
   updateSelectInput(
     session = getDefaultReactiveDomain(),
     "submission",
-    choices = c("", unique(reviews$submission))
+    choices = c("", unique(reviews()$submission))
   )
+  show_review_table(input, output, reviews, submission)
 
-  submission <- reactive({ input$submission })
+  observeEvent(input$refresh_comments, {
+    dccvalidator::with_busy_indicator_server("refresh_comments", {
+      reviews <<- reactive(pull_reviews_table(syn, reviews_table))
+      updateSelectInput(
+        session = getDefaultReactiveDomain(),
+        "submission",
+        choices = c("", unique(reviews()$submission)),
+        selected = submission()
+      )
+      show_review_table(input, output, reviews, submission)
+    })
+  })
+}
 
+#' @title Pull latest review table
+#' 
+#' @description Pull latest review table from Synapse.
+#' 
+#' @inheritParams mod_panel_section_server
+#' @keywords internal
+pull_reviews_table <- function(syn, reviews_table) {
+  reviews <- syn$tableQuery(glue::glue("SELECT * FROM {reviews_table}"))
+  reviews <- readr::read_csv(reviews$filepath) %>%
+    dplyr::mutate(scorer = get_display_name(syn, .data$scorer))
+  reviews
+}
+
+#' @title Show review table
+#'
+#' @description Show review table.
+#'
+#' @inheritParams mod_panel_section_server
+#' @param reviews Dataframe review table.
+#' @param submission Submission name.
```

Comment:
That's true, good catch. I can make it so `reviews` is not reactive, but `submission` will need to stay reactive due to this function not being called within an appropriate shiny context (i.e. `observe()`, etc). Would it be better to make `reviews` nonreactive (since it can be) and leave `submission` reactive, or have both be reactive for consistency?

Reply:
My vote would be to leave submission reactive (and update the parameter documentation accordingly), but make reviews nonreactive. To me it's clearer this way, but I don't feel super strongly about it.

Reply:
Sounds good. I don't have a strong opinion on it, either. Taken care of in latest commits.